editor/actions.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+export function savePost( dispatch, post ) {
```

Comment:
Already noted by @youknowriad that we'll need to consider adopting a pattern for asynchronous data flow. This file is perhaps a bit misleading since at a glance I'd expect it to include a set of action creators, but this isn't a true action creator in that it accepts `dispatch` and doesn't return an action object.

```
type ActionCreator = (...args: any) => Action | AsyncAction
```

http://redux.js.org/docs/Glossary.html#action-creator

I'm fine to flesh this out separately; `redux-thunk` is a popular option, but I've become less a fan of it over time, since thunks are hard to test, less predictable, and with the default implementation, encourage access to state via the second `getState` argument (thus breaking the unidirectional data flow).

If I were to suggest an alternative, I think it shouldn't be much trouble to devise a pattern to encourage treating actions with asynchronous effects as _expressing intent_. Using middleware(s), those intents could be mapped to side effects which themselves may incur additional actions.

Roughly something like: https://gist.github.com/aduth/b36ac70f0d3cd3f61faa0b2b81b98a2a

Not set in stone, specifically:

- Couldn't `savePost` just return `REQUEST` ? Maybe we don't need a separate effects mapping. But sometimes the save intent isn't always just about firing a request, but affecting reducer values (e.g. `SAVE_POST` provides semantic meaning that of just `REQUEST` which should cause the UI to update to reflect in-progress save)
- Instead of an effect's `success` property, we could enforce that the request completion itself should be an action expressing intent `REQUEST_COMPLETED` that has its own effect(s) to cause `receivePost` to be dispatched
- Maybe we forgo the `REQUEST` middleware and treat effects as mini-middlewares which have access to `dispatch` and could just use fetch or wp-api.js in their implementations

Anyways, this has been a bit of a brain dump that'll get lost in pull request history, but jotting down now that it's becoming relevant since this has been on my mind lately.

A few additional resources from side projects where I've been applying these ideas:

- https://github.com/aduth/dones/blob/master/src/state/requests/middleware.js
- https://github.com/aduth/dones/blob/master/src/state/dones/effects.js
- https://github.com/aduth/refx